sympy/tensor/array/dense_ndim_array.py

```diff
@@ -150,6 +150,8 @@ def __setitem__(self, index, value):
     def as_mutable(self):
         return MutableDenseNDimArray(self)
 
+    def _eval_simplify(self, **kwargs):
+        return self.applyfunc(simplify)
```

Comment:
I would rather recommend `self.applyfunc(lambda x: x._eval_simplify(**kwargs))`... `simplify` is the method available to end-users, `_eval_simplify` is for the internal code (`simplify` does a lot of parsing and error checking before calling `._eval_simplify`... once you have entered `_eval_simplify` it doesn't make any sense to call `simplify`).

Reply:
When returning `self.applyfunc(lambda x: x._eval_simplify(**kwargs))` it raises AttributeError.
```>>> from sympy.abc import x
>>> from sympy import sin, cos, simplify, Array
>>> A = Array([sin(2*x) - 2*sin(x)*cos(x), 0])
>>> simplify(A)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/uchiha/Desktop/sympy/sympy/simplify/simplify.py", line 559, in simplify
    return _eval_simplify(**kwargs)
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 154, in _eval_simplify
    return self.applyfunc(lambda x: x._eval_simplify(**kwargs))
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/ndim_array.py", line 292, in applyfunc
    return type(self)(map(f, Flatten(self)), self.shape)
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 129, in __new__
    return cls._new(iterable, shape, **kwargs)
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 138, in _new
    flat_list = flatten(flat_list)
  File "/home/uchiha/Desktop/sympy/sympy/utilities/iterables.py", line 75, in flatten
    for el in iterable:
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 154, in <lambda>
    return self.applyfunc(lambda x: x._eval_simplify(**kwargs))
AttributeError: 'Add' object has no attribute '_eval_simplify'
```

Reply:
Ping @Upabjojr

Reply:
@Upabjojr I may also clarify that not every sympy object has `_eval_simplify`.

Reply:
@Upabjojr  By looking at the code, it seems that `simplify` checks only one condition (that if `expr` being scalar zero) before calling `_eval_simplify` method of expression.